adsingestp/parsers/jats.py

```diff
@@ -913,22 +914,35 @@ def citation_context(
         self.back_meta = document.back
         body = document.body
         xrefs = body.find_all("xref")
+
+        # only include bibliographic references
+        cite_index = []
+        for index, xref in enumerate(xrefs):
+            if xref.get("ref-type", "") == "bibr":
+                cite_index.append(index)
+        xrefs = [xrefs[i] for i in cite_index]
+
         cites = {}  # {rid_1: ["context 1", "context 2", ...]}
         for x in xrefs:
             id = x["rid"]
             immediate_para = x.find_parent("p")  # try to find the containing paragraph
             if immediate_para:
-                context = immediate_para.get_text()
-                if len(context) < num_char:
-                    prev_para = immediate_para.find_previous_sibling("p")
-                    if prev_para:
-                        context = prev_para.get_text() + context
-                    next_para = immediate_para.find_next_sibling("p")
-                    if next_para:
-                        context = context + next_para.get_text()
+                context = immediate_para
+                if text_output:
```

Comment:
You're missing the `else` branch here, so if `text_output=False` and `immediate_para` exists, nothing will be returned.

Reply:
I believe this still should return beautifulsoup output if `text_output=False`, and it passes the test with that output.  This is because `context=immediate_para` on line 930 is the beautifulsoup object that will be returned and it is only altered if `text_output=True`.  I made this change to make it unnecessary to alter the `num_char=500` keyword.  If adding an `else` clause makes it more clear, I will make that change.